apps/heft/src/stages/BuildStage.ts

```diff
@@ -123,6 +123,20 @@ export interface ITypeScriptConfiguration extends ISharedTypeScriptConfiguration
   isLintingEnabled: boolean | undefined;
 }
 
+/**
+ * @public
+ */
+export interface IApiExtractorConfiguration {
+  /**
+   * If set to true, use the project's TypeScript compiler version for API Extractor's
+   * analysis. API Extractor's included TypeScript compiler can generally correctly
+   * analyze typings generated by older compilers, and referencing the project's compiler
+   * can cause issues. If issues are encountered with API Extractor's included compiler,
+   * set this option to true.
```

Comment:
Explain that this corresponds to API Extractor's `--typescript-compiler-folder` CLI parameter and `IExtractorInvokeOptions.typescriptCompilerFolder` API. This will dispel some mystery, since people can read the API Extractor docs for more detail.